addons/mail/sfu_server/src/client.js

```diff
@@ -0,0 +1,561 @@
+     * other call participants.
+     *
+     * @param {Object} info
+     * @param {Object} [param0]
+     * @param {boolean} [param0.needRefresh] true if the server should refresh the local info from all sessions of this channel
+     */
+    updateInfo(info, { needRefresh } = {}) {
+        this._info = info;
+        this._send({
+            name: CLIENT_MESSAGE.INFO_CHANGE,
+            payload: { info, needRefresh },
+        });
+    }
+
+    /**
+     * Stop or resume the consumption of tracks from the other call participants.
+     *
+     * @param {number} sessionId
+     * @param {Object} [param1]
+     * @param {Array<[string, boolean]>} [param1.states] e.g: [["audio", true], ["camera", false]]
+     * @param {boolean} [param1.disconnect] true if we want a complete disconnection from this session
+     */
+    updateDownload(sessionId, { states, disconnect } = {}) {
+        if (disconnect) {
+            this._removeConsumers(sessionId);
+            return;
+        }
+        for (const [type, active] of states) {
+            if (!SUPPORTED_TYPES.has(type)) {
+                throw new Error(`Unsupported media type ${type}`);
+            }
+            const consumer = this._consumers[type].get(sessionId);
+            if (consumer) {
+                if (active) {
+                    consumer.resume();
+                } else {
+                    consumer.pause();
+                }
+            }
+        }
+        this._send({
+            name: CLIENT_MESSAGE.CONSUMPTION_CHANGE,
+            payload: { sessionId, states },
+        });
+    }
+
+    /**
+     * @param {"audio" | "camera" | "screen"} type
+     * @param {MediaStreamTrack} [track] track to be sent to the other call participants,
+     * not setting it will remove the track from the server
+     */
+    async updateUpload(type, track) {
+        if (!SUPPORTED_TYPES.has(type)) {
+            throw new Error(`Unsupported media type ${type}`);
+        }
+        let producer = this._producers[type];
+        if (producer) {
+            if (track) {
+                await producer.replaceTrack({ track });
+            }
+            this._send({
+                name: CLIENT_MESSAGE.PRODUCTION_CHANGE,
+                payload: { type, active: Boolean(track) },
+            });
+            return;
+        }
+        if (!track) {
+            return;
+        }
+        try {
+            producer = await this._ctsTransport.produce({
+                ...PRODUCER_OPTIONS,
+                track,
+                appData: { type },
+            });
+        } catch (error) {
+            this.errors.push(error);
+            clearTimeout(this._recoverConsumerTimeouts[type]);
+            // if we reach the max error count, we restart the whole connection from scratch
+            if (this.errors.length > MAX_ERRORS) {
+                // not awaited
+                this._handleConnectionEnd();
+                return;
```

Comment:
the idea is that the error is handled internally (the recovery process is not outside of the sfuClient API)

Reply:
I suppose this can be observed from outside through the `stateChange` event so it's not like the user code can't react to what's going on and show something nice in the UI.